test/hotspot/jtreg/runtime/cds/appcds/loaderConstraints/DynamicLoaderConstraintsTest.java

```diff
@@ -55,6 +55,7 @@
 /**
  * @test id=custom-cl-zgc
  * @requires vm.cds.custom.loaders
+ * @requires vm.cds.default.archive.available
```

Comment:
Don't understand this one. This test is hard to follow but I think it does its own dumping of the CDS archive, so I don't see why it would need the default archive to exist. ???

Reply:
we run into this error without the jsa in the zgc case
runtime/cds/appcds/loaderConstraints/DynamicLoaderConstraintsTest.java#custom-cl-zgc

```
[0.002s][error][cds] Specified shared archive file not found (/jtreg_hotspot_tier2_work/JTwork/scratch/2/appcds-06h49m10s700.jsa)
[0.002s][error][cds] Not a valid shared archive file (/jtreg_hotspot_tier2_work/JTwork/scratch/2/appcds-06h49m10s700.jsa)
[0.002s][error][cds] CDS is incompatible with other specified options.
Error occurred during initialization of VM
Unable to use shared archive: invalid archive
```

Reply:
But that is not a default archive. ???

Reply:
The error is strange
```
[0.002s][error][cds] Specified shared archive file not found (/jtreg_hotspot_tier2_work/JTwork/scratch/2/appcds-06h49m10s700.jsa)
```
This is not the default archive file it cannot find. This is the generated one, no?

Reply:
If I regenerate the needed archive  (the custom-cl-zgc test needs classes_nocoops.jsa, I think because of ZGC)  by calling  `java -Xshare:dump -XX:-UseCompressedOops`  the test works nicely.
So adding  `vm.cds.default.archive.available`  is not really accurate because we miss the other archive.  It just works because in my test JVM we have BOTH the default archive classes.jsa and the classes_nocoops.jsa  removed.
Maybe we should have `vm.cds.nocoops.archive.available`  additionally to make clear what is really missing .
And probably better failure messages showing what's missing  would not hurt too, but I have a separate JBS issue for this.